errbot/backends/irc.py

```diff
@@ -481,8 +539,61 @@ def on_join(self, connection, event):
         # We need to wait to endofnames message.
         room_name = event.target
         with self._recently_joined_lock:
+            if room_name not in self.bot._rooms:
+                self.bot._rooms[room_name] = IRCRoom(room_name, self.bot)
             self._recently_joined_to.add(room_name)
 
+    def on_currenttopic(self, connection, event):
+        """
+            When you Join a room with a topic set this event fires up to
+            with the topic information.
+            If the room that you join don't have a topic set, nothing happens.
+            Here is NOT the place to fire the :meth:callback_room_topic event, for
+            that case exist on_topic.
+
+            :param:
+                connection: Is an 'irc.client.ServerConnection' object
+
+            :param:
+                event: Is an 'irc.client.Event' object
+                The event.arguments[0] contains the room name
+                The event.arguments[1] contains the topic of the room.
+        """
+        room_name, current_topic = event.arguments
+        self.bot._rooms[room_name].cb_set_topic(current_topic)
+
+    def on_topic(self, connection, event):
+        """
+            On response to the /TOPIC command if the room have a topic.
+            If the room don't have a topic the event fired is on_notopic
+            :param:
+                connection: Is an 'irc.client.ServerConnection' object
+
+            :param:
+                event: Is an 'irc.client.Event' object
+                The event.target contains the room name.
+                The event.arguments[0] contains the topic name
+        """
+        room_name = event.target
+        current_topic = event.arguments[0]
+        self.bot._rooms[room_name].cb_set_topic(current_topic)
+        self.bot.callback_room_topic(self.bot._rooms[room_name])
+
+    def on_notopic(self, connection, event):
+        """
+            This event fires ip when there is no topic set on a room
+
+            :param:
+                connection: Is an 'irc.client.ServerConnection' object
+
+            :param:
+                event: Is an 'irc.client.Event' object
+                The event.arguments[0] contains the room name
+        """
```

Comment:
This param syntax is wrong, these should go on the same line, as in `:param event:` otherwise Sphinx will fail to parse it.